test/calendar_test.exs

```diff
@@ -0,0 +1,210 @@
+if Code.ensure_loaded?(Calendar) do
+  defmodule CalendarTest do
+    use ExUnit.Case, async: true
+    import Postgrex.TestHelper
+    alias Postgrex, as: P
+
+    setup do
+      opts = [database: "postgrex_test", backoff_type: :stop,
+              extensions: [{Postgrex.Extensions.Calendar, []}]]
+      {:ok, pid} = P.start_link(opts)
+      {:ok, [pid: pid]}
+    end
+
+    test "decode time", context do
+      assert [[~T[00:00:00.000000]]] = query("SELECT time '00:00:00'", [])
+      assert [[~T[01:02:03.000000]]] = query("SELECT time '01:02:03'", [])
+      assert [[~T[23:59:59.000000]]] = query("SELECT time '23:59:59'", [])
+      assert [[~T[04:05:06.000000]]] = query("SELECT time '04:05:06 PST'", [])
+
+      assert [[~T[00:00:00.123000]]] = query("SELECT time '00:00:00.123'", [])
+      assert [[~T[00:00:00.123456]]] = query("SELECT time '00:00:00.123456'", [])
+      assert [[~T[01:02:03.123456]]] = query("SELECT time '01:02:03.123456'", [])
+    end
+
+    test "decode timetz", context do
+      assert [[~T[00:00:00.000000]]] = query("SELECT time with time zone '00:00:00 UTC'", [])
+      assert [[~T[01:02:03.000000]]] = query("SELECT time with time zone '01:02:03 UTC'", [])
+      assert [[~T[23:00:00.000000]]] = query("SELECT time with time zone '00:00:00+1'", [])
```

Comment:
@lau here is an example of a negative `microsec` as we need to go back 1 hour from `0 microseconds` to get UTC.